projects/vdk-plugins/vdk-impala/tests/conftest.py

```diff
@@ -39,18 +38,31 @@ def docker_compose_file(pytestconfig):
         VDK_IMPALA_PORT: "21050",
     },
 )
-def impala_service(docker_ip, docker_services):
+def impala_service(request):
     """Ensure that Impala service is up and responsive."""
-    runner = CliEntryBasedTestRunner(impala_plugin)
-
-    # give the server some time to start before checking if it is ready
-    # before adding this sleep there were intermittent fails of the CI/CD with error:
-    # requests.exceptions.ConnectionError:
-    #   ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
-    # More info: https://stackoverflow.com/questions/383738/104-connection-reset-by-peer-socket-error-or-when-does-closing-a-socket-resu
-    time.sleep(3)
-
-    docker_services.wait_until_responsive(
-        timeout=120.0, pause=1, check=lambda: _is_responsive(runner)
-    )
-    time.sleep(10)
+    port = int(os.environ[VDK_IMPALA_PORT])
+    container = DockerContainer("cpcloud86/impala:latest").with_bind_ports(port, port)
+
+    try:
+        container.start()
+        runner = CliEntryBasedTestRunner(impala_plugin)
+        wait_for(partial(_is_responsive, runner))
+        # wait 2 seconds to make sure the service is up and responsive
+        # might be unnecessary but it's out of abundance of caution
+        time.sleep(2)
+        print(
```

Comment:
You should log this instead of printing.

Reply:
Yes, I agree. But as I am refactoring code. I am not 100% sure why we are using print instead of log. There should be a good reason.  I need to research that and would do that in a separate PR.